app/src/main/java/org/pesmypetcare/mypetcare/utilities/DateTime.java

```diff
@@ -0,0 +1,182 @@
+    private static final int MAX_MINUTES_SECONDS = 60;
+    private static final int FIRST_TWO_DIGITS = 10;
+    private static final int MONTH_START_POS = 5;
+    private static final int DAY_START_POS = 8;
+    private static final int HOUR_END_POS = 13;
+    private static final int MIN_START_POS = 14;
+    private static final int MIN_END_POS = 16;
+    private static final int SEC_START_POS = 17;
+    private int year;
+    private int month;
+    private int day;
+    private int hour;
+    private int minutes;
+    private int seconds;
+
+    public DateTime(int year, int month, int day, int hour, int minutes, int seconds) throws InvalidFormatException {
+        this.year = year;
+        this.month = month;
+        this.day = day;
+        this.hour = hour;
+        this.minutes = minutes;
+        this.seconds = seconds;
+        if (isNegative(year, month, hour, minutes, seconds) || isOutOfRange(month, hour, minutes, seconds)) {
+            throw new InvalidFormatException();
+        }
+    }
+
+    public DateTime(String dateTime) {
+        this.year = Integer.parseInt(dateTime.substring(0, LEAP_YEAR_FREQ));
+        this.month = Integer.parseInt(dateTime.substring(MONTH_START_POS, JULY));
+        this.day = Integer.parseInt(dateTime.substring(DAY_START_POS, FIRST_TWO_DIGITS));
+        this.hour = Integer.parseInt(dateTime.substring(HOUR_START_POS, HOUR_END_POS));
+        this.minutes = Integer.parseInt(dateTime.substring(MIN_START_POS, MIN_END_POS));
+        this.seconds = Integer.parseInt(dateTime.substring(SEC_START_POS));
+    }
+
+    private boolean isOutOfRange(int month, int hour, int minutes, int seconds) {
+        int nDays = numberOfDays(year, month);
+        boolean dateOutOfRange = month > DECEMBER || day > nDays;
+        boolean hourOutOfRange = hour >= MAX_HOUR || minutes >= MAX_MINUTES_SECONDS || seconds > MAX_MINUTES_SECONDS;
+        return dateOutOfRange || hourOutOfRange;
+    }
+
+    private boolean isNegative(int year, int month, int hour, int minutes, int seconds) {
+        boolean isYearNegative = year < 0 || month < 0 || day < 0;
+        boolean isDateNegative = hour < 0 || minutes < 0 || seconds < 0;
+        return isYearNegative || isDateNegative;
+    }
+
+    private int numberOfDays(int year, int month) {
+        if (month <= JULY) {
+            if (month != FEBRUARY) {
+                if (month % 2 == 0) {
+                    return DAYS_30;
+                } else {
+                    return DAYS_31;
+                }
+            } else if (isLeapYear(year)) {
+                return DAYS_29;
+            } else {
+                return DAYS_28;
+            }
+        } else if (month % 2 == 0) {
+            return DAYS_31;
+        }
+        return DAYS_30;
+    }
+
+    private boolean isLeapYear(int year) {
+        return year % LEAP_YEAR_FREQ == 0 && (year % TWO_LAST_DIGITS != 0
+            || (year / TWO_LAST_DIGITS) % LEAP_YEAR_FREQ == 0);
+    }
+
+    @Override
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [NCSS for this method is 19 (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5311927)